File: Modules/Common/AbnormalInfo/SIE.Web.AbnormalInfo/AbnormalInfos/Commands/AbnormalInfoDefHandlerSelectCommand.js
SIE.defineCommand('SIE.Web.AbnormalInfo.AbnormalInfos.Commands.AbnormalInfoDefHandlerSelectCommand',
    {
        extend: 'SIE.cmd.LookupCommandBase',
        userConfig: {
            dataParams: {
                specKeyPrototyName: 'HandlerId',
                targetClassName: 'SIE.Resources.Employees.Employee'
            }
        },
        meta: { text: "选择", group: "edit", iconCls: "icon-PlaylistCheck icon-blue" },
        /**
         * override 保存方法
         * @param {} win 
         * @returns {} 
         */
        save: function (win) {
            var me = this;
            var indata = {};
            var selections = this._targetView.getSelection();
            if (selections && selections.length > 0) {
                var operationDatas = [];
                SIE.each(selections,
                    function (item) {
                        var userId = item.getId();
                        if (me._sourceViewSelectItems.indexOf(userId) === -1) {
                            var defHandlers = {
                                AbnormalInfoDefinitionId: me._sourceId,
                                HandlerId: userId
                            };
                            operationDatas.push(defHandlers);
                        }

                    });
                indata = operationDatas;
                me._targetView.execute({
                    data: indata,
                    success: function (res) {
                        win.close(); //关闭模态窗口
                        me._ownerView.loadChildData(true); //重载视图数据
                    }
                },
                    me._ownerView);
            } else {
                SIE.Msg.showWarning('没有可提交的数据'.t());
            }
        }

    });

File: Modules/Common/AbnormalInfo/SIE.Web.AbnormalInfo/AbnormalInfos/Commands/AbnormalInfoDefHandlerSelectCommand.cs
using SIE.Domain;
using SIE.AbnormalInfo.AbnormalInfos;
using SIE.Web.Command;
using System;
using System.Collections.Generic;

namespace SIE.Web.AbnormalInfo.AbnormalInfos.Commands
{
    /// <summary>
    /// 异常信息定义选择处理人按钮
    /// </summary>
    public class AbnormalInfoDefHandlerSelectCommand : ViewCommand
    {
        /// <summary>
        /// 执行方法
        /// </summary>
        /// <param name="args"></param>
        /// <param name="scope"></param>
        /// <returns></returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            var meta = SIE.ClientEntities.Find(args.Type);
            var savedData = RepositoryFactory.Find(meta.EntityType).NewList();
            var list = args.Data.ToJsonObject<List<AbnormalInfoDefinitionEmployee>>();    //转换数据成实体
            Check.NotNullOrEmpty(list, nameof(list));       //检验非空

            if (list == null || list.Count == 0)
                throw new ArgumentNullException("{0}数据参数不能为空".L10nFormat(nameof(list)));

            foreach (var item in list)
            {
                var tempInsItem = new AbnormalInfoDefinitionEmployee();
                tempInsItem.HandlerId = item.HandlerId;
                tempInsItem.AbnormalInfoDefinitionId = item.AbnormalInfoDefinitionId;
                savedData.Add(tempInsItem);
            }

            RF.Save(savedData);

            return true;
        }
    }
}
